Browser_IDE/executionEnvironment_Page.js

```diff
@@ -80,34 +91,57 @@ document.getElementById("canvas").addEventListener("click", async function () {
 
 // Convenience function for reporting errors, printing them to the terminal
 // and also sending a message to the main window.
-function ReportError(block, message, line, formatted=false){
+function ReportError(block, message, line, stacktrace ,formatted=false){
     let outputMessage = message != "";
+    let stackTrace = stacktrace;
+
+    // Ensure block and message are strings
+    block = block || "";
+    message = message || "";
+
+    // Escape only the user-provided input
+    let escapedBlock = escapeHtml(block);
+    let escapedMessage = escapeHtml(message);
+
+    if (outputMessage && line != null && !formatted){
+        escapedMessage = "Error on line "+line+": "+escapedMessage;
 
-    if (outputMessage && line != null)
-        message = "Error on line "+line+": "+message;
+    }
+
+    
 
-    if (block != null && block != "" && block != "__USERCODE__null") {
-        if (!block.startsWith(userCodeBlockIdentifier)){
-            message = "Please file a bug report and send us the following info!\n    Error in file: "+block+"\n    "+message;
-            block = "Internal Error";
+    if (escapedBlock != null && escapedBlock != "" && escapedBlock != "__USERCODE__null") {
+        if (!escapedBlock.startsWith(userCodeBlockIdentifier)){
+            escapedMessage = "Please file a bug report and send us the following info!\n    Error in file: "+escapedBlock+"\n    "+escapedMessage;
+            escapedBlock = "Internal Error";
         }
         else{
-            block = block.slice(userCodeBlockIdentifier.length);
+            escapedBlock = escapedBlock.slice(userCodeBlockIdentifier.length);
         }
         if (outputMessage)
-            message = "(" + block + ") " + message;
+            escapedMessage = "(" + escapedBlock + ") " + escapedMessage;
+        
+    }
+    escapedMessage = "\x1b[0m\x1b[31m" + escapedMessage + "\x1b[0m";

```

Comment:
Putting this here breaks stack collapsing later on:
![image](https://github.com/thoth-tech/SplashkitOnline/assets/42032199/0b5cd78b-5fbb-437c-824b-540a1c312e3a)
The message+stack now appears on another line to the arrow and can't be closed.

Just remove this line (`escapedMessage = "\x1b[0m\x1b[31m" + escapedMessage + "\x1b[0m";`), and put it as an `else` block below like in my previous suggestion 😃